File: jison/code.c
#include <stdio.h>
#include <string.h>

int test(char* s, int length) {
    int i = 0;
	int j = length;
    j--;
    while (i < j) {
        if (s[i] != s[j]) {
			return 0;
        }
        i++;
        j--;
    }
    return 1;
}

int main() {
    char s[200];
    printf("Enter the string to test: ");
    scanf("%s", s);
    if (test(s, strlen(s))) {
		printf("true");
    }
    else {
		printf("false");
    }
    return 0;
}

File: jison/code.py
def test(s, length): 
    i = 0
    j = length
    j -= 1
    while i < j :
        if s[i] != s[j]:
            return 0
        i += 1
        j -= 1
    return 1

def main(): 
    s = ""
    print("Enter the string to test: ")
    s = input()
    if test(s, len(s)):
        print("true")
    else:
        print("false")
    return 0

main()
